### ISSUE TITLE
Exception when using bounds expressions containing other struct members in structs

### ISSUE THREAD

**Issue by lenary** [Microsoft/checkedc-clang#244]
I think we inevitably knew our code would break for this.

If you specify a struct like the following:
```
struct S1 {
  int size;
  array_ptr<int> items : count(size);
}
```

Then clang will hit the [`llvm_unreachable(...)` at the end of EmitDeclRefLValue](https://github.com/Microsoft/checkedc-clang/blob/master/lib/CodeGen/CGExpr.cpp#L2215) when generating the bounds for an expression like `s1->items[i]`.

Not really sure what we can do here before we fix up inferred bounds on structs.

**Comment by dtarditi** [comment]
This still needs to be implemented; see issue #249.

**Comment by dtarditi** [comment]
Inferred bounds for structs were implemented by #258.
